eng/pipelines/typespec-requirement.yml

```diff
@@ -0,0 +1,14 @@
+trigger: none
```

Comment:
I assume the plan is to have this trigger on all branches?

Reply:
`trigger` is for CI, so `none` should be correct.  This pipeline only applies to PRs, it doesn't make sense to run against say `main` directly, since it only looks at files changed in a PR.

For PRs, yes I think we would trigger on all branches, which I believe is the default if you don't specify.

I believe this pipeline should have exactly the same triggers as existing pipeline `TypeSpec Validation`.  I'm planning to copy both the YML and DevOps settings from this existing pipeline.